### ISSUE TITLE
Flip doesn't work for Black Box IOs with Clocks

### ISSUE THREAD

**Issue by mwachs5** [freechipsproject/chisel3#282]
This is perhaps related to #249 

If I create an IO Bundle with a Clock type, and then create a BlackBox with that IO.flip(), it doesn't get flipped in the black box:

Chisel Input:

```

class FooIO extends Bundle {
  val fooData = Bool(INPUT)
  val fooClk = Clock(INPUT)
}

class MyFooBlackBox extends BlackBox {
  val io = new FooIO().flip()
}
```

Creates FIRRTL Output:

```
extmodule MyFooBlackBox :
  input fooClk : Clock
  output fooData : UInt<1>
```

This causes problems later when Chisel tries to connect to the blackbox.

**Comment by sdtwigg** [comment]
I want to say this is a manifestation of the direction bugs discussed in #440

**Comment by ducky64** [comment]
Will try to fix with #440

**Comment by ducky64** [comment]
Ok, looked into this and the problem is that the top-level `io` object doesn't actually get emitted for a BlackBox, instead its members become the top-level IO of a BlackBox and anything that applied to the `io` container itself is dropped.

This particular issue can be handled through special cases (like passing flipped, or even the user direction into `DefBlackbox` and having things emit accordingly), but if we start generating FIRRTL bulk-connects, then this splitting becomes a major issue since the `io` object *no longer exists*.
